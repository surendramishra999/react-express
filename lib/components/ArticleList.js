import React from 'react';
import Article from './Article';

const ArticleList=(props)=>{
  return (
    <div>
      {Object.values(props.articles).map(article =>
        <Article
          key={article.id}
          article={article}
          authors={props.authors}
        />
      )}
    </div>
  );
};

export default ArticleList;