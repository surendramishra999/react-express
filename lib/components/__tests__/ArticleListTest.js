import React from 'react';
import ArticleList from '../ArticleList';

import renderer from 'react-test-renderer';


describe('Article List rendering tests', () => {
  const testProps={
    articles:{
      a:{id:'a'},
      b:{id:'b'}
    },
    store:{
      lookupAuthor: jest.fn(()=>({})),
    },
  };

  
  it('should behave like tree', () => {
    const tree=renderer.create(
      <ArticleList
        {...testProps}  
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  
});

