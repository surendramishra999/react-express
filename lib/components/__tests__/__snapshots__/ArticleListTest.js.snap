// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article List rendering tests should behave like tree 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <ArticleContainer
        article={
            Object {
                "body": "",
                "date": "",
                "id": "a",
                "title": "",
              }
        }
    />
    <ArticleContainer
        article={
            Object {
                "body": "",
                "date": "",
                "id": "b",
                "title": "",
              }
        }
    />
</div>,
  "nodes": Array [
    <div>
      <ArticleContainer
            article={
                  Object {
                        "body": "",
                        "date": "",
                        "id": "a",
                        "title": "",
                      }
            }
      />
      <ArticleContainer
            article={
                  Object {
                        "body": "",
                        "date": "",
                        "id": "b",
                        "title": "",
                      }
            }
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_currentlyRenderingComponent": null,
    "_didScheduleRenderPhaseUpdate": false,
    "_dispatcher": Object {
      "readContext": [Function],
      "useCallback": [Function],
      "useContext": [Function],
      "useDebugValue": [Function],
      "useEffect": [Function],
      "useImperativeHandle": [Function],
      "useLayoutEffect": [Function],
      "useMemo": [Function],
      "useReducer": [Function],
      "useRef": [Function],
      "useState": [Function],
    },
    "_element": <ArticleList
      articles={
            Object {
                  "a": Object {
                    "body": "",
                    "date": "",
                    "id": "a",
                    "title": "",
                  },
                  "b": Object {
                    "body": "",
                    "date": "",
                    "id": "b",
                    "title": "",
                  },
                }
      }
/>,
    "_firstWorkInProgressHook": null,
    "_forcedUpdate": false,
    "_instance": null,
    "_isReRender": false,
    "_newState": null,
    "_numberOfReRenders": 0,
    "_previousComponentIdentity": [Function],
    "_renderPhaseUpdates": null,
    "_rendered": <div>
      <ArticleContainer
            article={
                  Object {
                        "body": "",
                        "date": "",
                        "id": "a",
                        "title": "",
                      }
            }
      />
      <ArticleContainer
            article={
                  Object {
                        "body": "",
                        "date": "",
                        "id": "b",
                        "title": "",
                      }
            }
      />
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "_workInProgressHook": null,
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ArticleList
    articles={
        Object {
            "a": Object {
              "body": "",
              "date": "",
              "id": "a",
              "title": "",
            },
            "b": Object {
              "body": "",
              "date": "",
              "id": "b",
              "title": "",
            },
          }
    }
/>,
}
`;
